src/mon/MDSMonitor.cc

```diff
@@ -621,9 +629,20 @@ void MDSMonitor::_updated(MMDSBeacon *m)
   mon->clog->info() << m->get_orig_source_inst() << " "
 	  << ceph_mds_state_name(m->get_state()) << "\n";
 
-  if (m->get_state() == MDSMap::STATE_STOPPED) {
+  switch (m->get_state()) {
+  case MDSMap::STATE_STOPPED:
+  case MDSMap::STATE_DAMAGED:
```

Comment:
We do this for STOPPED ranks so that they know to proceed from STOPPED to DNE (and thereby respawn themselves).

For DAMAGED ranks this isn't necessary, because they explicitly do a wait for the beacon response, and if it doesn't come they will kill themselves.  They never need to see the MDS map again.